### ISSUE TITLE
Tray icons become stuck when clicking on them

### ISSUE THREAD

**Issue by evilzer** [polybar/polybar#1440]
Noticed this problem after updating to the latest version. Don't remember ever seeing before that...

If I click on a icon in the tray, that icon will become "stuck", if the icon changes after that, both the old and new icon will be displayed over the same area. This only happens if you interact with the icon, and only that icon is affected. If the icon changes without clicking on them it works as intended.

**Normal:**
![2018-09-25-172330_1366x28_scrot](https://user-images.githubusercontent.com/3471368/46024899-0003e100-c0e8-11e8-8604-2f63bd7e7712.png)

**WiFi and disconnected show at the same time:**
![2018-09-25-172352_1366x28_scrot](https://user-images.githubusercontent.com/3471368/46024954-114ced80-c0e8-11e8-8560-4c5cf008f0fa.png)

No errors are output is generated.
WM: bspwm
```
polybar 3.2.1

Features: +alsa +curl +i3 +mpd +network +pulseaudio +xkeyboard

X extensions: +randr (+monitors) -render -damage -sync -composite +xkb +xrm +xcursor

Build type: RelWithDebInfo
Compiler: /usr/bin/c++
Compiler flags: -march=native -O2 -pipe -fstack-protector-strong -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```

**Comment by evilzer** [comment]
I just saw #1143, probably related.
Also just discovered that you don't have to click the icon to trigger the bug, just hovering over it with the mouse a short while is enough to trigger it.

**Comment by NBonaparte** [comment]
Does #1415 help at all?

**Comment by evilzer** [comment]
No, the patch did not change the behavior. It happens independent of what background/transparency setting is used.
But now every time the tray is drawn at startup this error comes up:
`error: Failed to store slice of root pixmap (XCB_DRAWABLE (9))`
And when a new icon is added:
`error: Failed to reconfigure tray background (XCB_DRAWABLE (9))`

**Comment by wioo** [comment]
Looks like the problem is with gtk3 package. I installed gtk 3.22.30 on arch linux (current 3.24.1+8+g220f77d8c1) and it works without a problem.

**Comment by evilzer** [comment]
This issue is fixed with gtk3 version 3.24.3.
